api/v1alpha3/azuremachine_validation.go

```diff
@@ -102,6 +103,61 @@ func ValidateOSDisk(osDisk OSDisk, fieldPath *field.Path) field.ErrorList {
 
 	allErrs = append(allErrs, validateStorageAccountType(osDisk.ManagedDisk.StorageAccountType, fieldPath)...)
 
+	if errs := ValidateManagedDisk(osDisk.ManagedDisk, osDisk.ManagedDisk, fieldPath.Child("managedDisk")); len(errs) > 0 {
```

Comment:
One more comment (noticed this from looking at #652), it seems a bit awkward to set the "old" object as itself for a create and it's different from what CAPI does, which set old to `nil` for validateCreate() https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha3/machine_webhook.go#L119

Reply:
ValidateOSDisk should take old and new and pass it through to `ValidateManagedDisk`, this makes the assumption that it's always only used in Create validation.

Reply:
the capi approach doesn't work for non-pointer values, I think? will try and see what I can do

Reply:
ok, if there's a good reason for passing in a non-nil old object in create (looks like there is) then 👍 

Still think we should pass through the old object from ValidateCreate/ValidateUpdate though.

Reply:
yeah, actually -- I don't really see a better way to handle updates for non-nil structs and reusing the same func. We can have two different funcs?

Reply:
I think this > two different funcs

would prefer avoiding copy paste logic

Reply:
I guess we can define validation funcs on the structs all the way down, and then call new.validate(old) at each layer?